diff --git a/config/glove80.keymap b/config/glove80.keymap
index bef4ddd..9009c3d 100644
--- a/config/glove80.keymap
+++ b/config/glove80.keymap
@@ -669,6 +669,15 @@ macros {
         tap-ms = <SELECT_WORD_DELAY>;
         bindings = <&kp DOWN &kp UP &kp RIGHT &kp LEFT>;
     )
+    screen_shot: screen_shot {
+            compatible = "zmk,behavior-macro";
+            #binding-cells = <0>;
+            bindings
+                = <&macro_press &kp LALT &kp LCTRL &kp LSHFT>
+                , <&macro_tap &kp PSCRN>
+                , <&macro_release &kp LALT &kp LCTRL &kp LSHFT>
+                ;
+        };
 
     //
     // select a word (jumps to next word upon each successive invocation)
@@ -2595,23 +2604,23 @@ macros {
 
         layer_Engrammer {
             bindings = <
-               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                          &kp F6                &kp F7               &kp F8               &kp F9               &kp F10
+               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                          &kp LS(INS)                &kp F7               &kp F8               &kp F9               &kp F10
             &kp EQUAL              &kp N1              &kp N2               &kp N3                  &kp N4     &kp N5                                                                                                                                      &kp N6                   &kp N7                &kp N8               &kp N9               &kp N0              &kp BSLH
             &kp GRAVE               &kp B  &homey_left RALT Y                &kp O                   &kp U    &kp SQT                                                                                                                                    &kp SEMI                    &kp L                 &kp D  &homey_right RALT W                &kp V                 &kp Z
                &cappy  &homey_left LGUI C  &homey_left LALT I  &homey_left LCTRL E     &shift_left LSHFT A  &kp COMMA                                                                                                                                     &kp DOT     &shift_right LSHFT H  &homey_right LCTRL T  &homey_right LALT S  &homey_right LGUI N                 &kp Q
-            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K  &kp MINUS      &thumb LAYER_Function ESC                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &kp FSLH                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
- &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END             &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number DELETE    &lower    &lower  &thumb LAYER_Mouse TAB  &space LAYER_Symbol SPACE            &thumb LAYER_World LPAR              &kp LBKT             &kp RBKT             &kp RPAR  &magic LAYER_Magic 0
+            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K  &kp MINUS      &thumb LAYER_Function DELETE                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &kp FSLH                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
+ &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END             &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number ESC    &lower    &lower  &thumb LAYER_Mouse ENTER  &space LAYER_Symbol SPACE            &thumb LAYER_World TAB              &kp LBKT             &kp RBKT             &screen_shot  &magic LAYER_Magic 0
             >;
         };
 
         layer_Engram {
             bindings = <
-               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                                      &kp F6                &kp F7               &kp F8               &kp F9               &kp F10
+               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                                      &kp LS(INS)                &kp F7               &kp F8               &kp F9               &kp F10
              &kp LBKT           &engram_1           &engram_2            &engram_3               &engram_4      &engram_5                                                                                                                                           &engram_6                &engram_7             &engram_8            &engram_9            &engram_0              &kp RBKT
            &engram_at               &kp B  &homey_left RALT Y                &kp O                   &kp U   &engram_apos                                                                                                                                      &engram_dquote                    &kp L                 &kp D  &homey_right RALT W                &kp V                 &kp Z
                &cappy  &homey_left LGUI C  &homey_left LALT I  &homey_left LCTRL E     &shift_left LSHFT A  &engram_comma                                                                                                                                      &engram_period     &shift_right LSHFT H  &homey_right LCTRL T  &homey_right LALT S  &homey_right LGUI N                 &kp Q
-            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K      &kp MINUS      &thumb LAYER_Function ESC                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &engram_question                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
- &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END                 &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number DELETE    &lower    &lower  &thumb LAYER_Mouse TAB  &space LAYER_Symbol SPACE                    &thumb LAYER_World LPAR          &engram_hash        &engram_slash             &kp RPAR  &magic LAYER_Magic 0
+            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K      &kp MINUS      &thumb LAYER_Function DELETE                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &engram_question                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
+ &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END                 &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number ESC    &lower    &lower  &thumb LAYER_Mouse ENTER  &space LAYER_Symbol SPACE                    &thumb LAYER_World TAB          &engram_hash        &engram_slash             &screen_shot  &magic LAYER_Magic 0
             >;
         };
 
@@ -2650,12 +2659,12 @@ macros {
 
         layer_QWERTY {
             bindings = <
-               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                     &kp F6                &kp F7               &kp F8                  &kp F9               &kp F10
+               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                     &kp LS(INS)                &kp F7               &kp F8                  &kp F9               &kp F10
             &kp EQUAL              &kp N1              &kp N2               &kp N3                  &kp N4  &kp N5                                                                                                                                    &kp N6                   &kp N7                &kp N8               &kp N9                  &kp N0              &kp BSLH
             &kp GRAVE               &kp Q  &homey_left RALT W                &kp E                   &kp R   &kp T                                                                                                                                     &kp Y                    &kp U                 &kp I                &kp O                   &kp P             &kp MINUS
                &cappy  &homey_left LGUI A  &homey_left LALT S  &homey_left LCTRL D     &shift_left LSHFT F   &kp G                                                                                                                                     &kp H     &shift_right LSHFT J  &homey_right LCTRL K  &homey_right LALT L  &homey_right LGUI SEMI               &kp SQT
-            &kp LSHFT               &kp Z               &kp X                &kp C                   &kp V   &kp B      &thumb LAYER_Function ESC                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER   &kp N                    &kp M             &kp COMMA              &kp DOT                &kp FSLH             &kp RSHFT
- &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END          &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number DELETE    &lower    &lower  &thumb LAYER_Mouse TAB  &space LAYER_Symbol SPACE          &thumb LAYER_World LPAR              &kp LBKT             &kp RBKT                &kp RPAR  &magic LAYER_Magic 0
+            &kp LSHFT               &kp Z               &kp X                &kp C                   &kp V   &kp B      &thumb LAYER_Function DELETE                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER   &kp N                    &kp M             &kp COMMA              &kp DOT                &kp FSLH             &kp RSHFT
+ &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END          &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number ESC    &lower    &lower  &thumb LAYER_Mouse ENTER  &space LAYER_Symbol SPACE          &thumb LAYER_World TAB              &kp LBKT             &kp RBKT                &screen_shot  &magic LAYER_Magic 0
             >;
         };
 
@@ -2764,8 +2773,8 @@ macros {
  &trans  &trans  &trans  &trans  &trans  &trans                                                                               &trans  &trans  &trans  &trans  &trans  &trans
  &trans  &trans   &kp Y  &trans  &trans  &trans                                                                               &trans  &trans  &trans   &kp W  &trans  &trans
  &trans   &kp C   &kp I   &kp E   &kp A  &trans                                                                               &trans   &kp H   &kp T   &kp S   &kp N  &trans
- &trans  &trans  &trans  &trans  &trans  &trans   &kp ESC   &trans             &trans             &trans   &trans    &kp RET  &trans  &trans  &trans  &trans  &trans  &trans
- &trans  &trans  &trans  &trans  &trans          &kp BSPC  &kp DEL  &tog LAYER_Typing  &tog LAYER_Typing  &kp TAB  &kp SPACE          &trans  &trans  &trans  &trans  &trans
+ &trans  &trans  &trans  &trans  &trans  &trans   &kp DEL   &trans             &trans             &trans   &trans    &kp RET  &trans  &trans  &trans  &trans  &trans  &trans
+ &trans  &trans  &trans  &trans  &trans          &kp BSPC  &kp ESC  &tog LAYER_Typing  &tog LAYER_Typing  &kp RET  &kp SPACE          &trans  &trans  &trans  &trans  &trans
             >;
         };
 
diff --git a/glove80.uf2 b/glove80.uf2
new file mode 100644
index 0000000..9605897
Binary files /dev/null and b/glove80.uf2 differ
diff --git a/glove80.uf2.zip b/glove80.uf2.zip
new file mode 100644
index 0000000..8866b12
Binary files /dev/null and b/glove80.uf2.zip differ
diff --git a/thumb.patch b/thumb.patch
new file mode 100644
index 0000000..79d01c2
--- /dev/null
+++ b/thumb.patch
@@ -0,0 +1,79 @@
+diff --git a/config/glove80.keymap b/config/glove80.keymap
+index d958570..2f01ef4 100644
+--- a/config/glove80.keymap
++++ b/config/glove80.keymap
+@@ -738,6 +738,16 @@ macros {
+         bindings = <&kp DOWN &kp UP &kp RIGHT &kp LEFT>;
+     )
+ 
++    screen_shot: screen_shot {
++            compatible = "zmk,behavior-macro";
++            #binding-cells = <0>;
++            bindings
++                = <&macro_press &kp LALT &kp LCTRL>
++                , <&macro_tap &kp PSCRN >
++                , <&macro_release &kp LALT &kp LCTRL>
++                ;
++        };
++
+     //
+     // select a word (jumps to next word upon each successive invocation)
+     //
+@@ -2655,7 +2665,7 @@ macros {
+ /*HACK*//{
+ };
+ 
+-    /* Automatically generated keymap */
++    /* Automatically generated key map */
+ / {
+     keymap {
+         compatible = "zmk,keymap";
+@@ -2663,12 +2673,12 @@ macros {
+ 
+         layer_Engrammer {
+             bindings = <
+-               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                          &kp F6                &kp F7               &kp F8               &kp F9               &kp F10
++               &kp F1              &kp F2              &kp F3               &kp F4                  &kp F5                                                                                                                                                                          &kp LS(INS)                &kp F7               &kp F8               &kp F9               &kp F10
+             &kp EQUAL              &kp N1              &kp N2               &kp N3                  &kp N4     &kp N5                                                                                                                                      &kp N6                   &kp N7                &kp N8               &kp N9               &kp N0              &kp BSLH
+             &kp GRAVE               &kp B  &homey_left RALT Y                &kp O                   &kp U    &kp SQT                                                                                                                                    &kp SEMI                    &kp L                 &kp D  &homey_right RALT W                &kp V                 &kp Z
+                &cappy  &homey_left LGUI C  &homey_left LALT I  &homey_left LCTRL E     &shift_left LSHFT A  &kp COMMA                                                                                                                                     &kp DOT     &shift_right LSHFT H  &homey_right LCTRL T  &homey_right LALT S  &homey_right LGUI N                 &kp Q
+-            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K  &kp MINUS      &thumb LAYER_Function ESC                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &kp FSLH                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
+- &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji ESC             &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number DELETE    &lower    &lower  &thumb LAYER_Mouse TAB  &space LAYER_Symbol SPACE            &thumb LAYER_World ENTER              &kp LBKT             &kp RBKT             &kp END  &magic LAYER_Magic 0
++            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K  &kp MINUS      &thumb LAYER_Function DELETE                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &kp FSLH                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
++ &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END             &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number  ESC   &lower    &lower  &thumb LAYER_Mouse ENTER  &space LAYER_Symbol SPACE            &thumb LAYER_World TAB              &kp LBKT             &screen_shot             &kp END  &magic LAYER_Magic 0
+             >;
+         };
+ 
+@@ -2678,8 +2688,8 @@ macros {
+              &kp LBKT           &engram_1           &engram_2            &engram_3               &engram_4      &engram_5                                                                                                                                           &engram_6                &engram_7             &engram_8            &engram_9            &engram_0              &kp RBKT
+            &engram_at               &kp B  &homey_left RALT Y                &kp O                   &kp U   &engram_apos                                                                                                                                      &engram_dquote                    &kp L                 &kp D  &homey_right RALT W                &kp V                 &kp Z
+                &cappy  &homey_left LGUI C  &homey_left LALT I  &homey_left LCTRL E     &shift_left LSHFT A  &engram_comma                                                                                                                                      &engram_period     &shift_right LSHFT H  &homey_right LCTRL T  &homey_right LALT S  &homey_right LGUI N                 &kp Q
+-            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K      &kp MINUS      &thumb LAYER_Function ESC                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &engram_question                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
+- &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji ESC                 &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number DELETE    &lower    &lower  &thumb LAYER_Mouse TAB  &space LAYER_Symbol SPACE                    &thumb LAYER_World ENTER          &engram_hash        &engram_slash             &kp END  &magic LAYER_Magic 0
++            &kp LSHFT               &kp G               &kp X                &kp J                   &kp K      &kp MINUS      &thumb LAYER_Function DELETE                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER  &engram_question                    &kp R                 &kp M                &kp F                &kp P             &kp RSHFT
++ &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END                 &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number ESC    &lower    &lower  &thumb LAYER_Mouse ENTER  &space LAYER_Symbol SPACE                    &thumb LAYER_World TAB          &engram_hash        &engram_slash             &kp screen_shot  &magic LAYER_Magic 0
+             >;
+         };
+ 
+@@ -2722,8 +2732,8 @@ macros {
+             &kp EQUAL              &kp N1              &kp N2               &kp N3                  &kp N4  &kp N5                                                                                                                                    &kp N6                   &kp N7                &kp N8               &kp N9                  &kp N0              &kp BSLH
+             &kp GRAVE               &kp Q  &homey_left RALT W                &kp E                   &kp R   &kp T                                                                                                                                     &kp Y                    &kp U                 &kp I                &kp O                   &kp P             &kp MINUS
+                &cappy  &homey_left LGUI A  &homey_left LALT S  &homey_left LCTRL D     &shift_left LSHFT F   &kp G                                                                                                                                     &kp H     &shift_right LSHFT J  &homey_right LCTRL K  &homey_right LALT L  &homey_right LGUI SEMI               &kp SQT
+-            &kp LSHFT               &kp Z               &kp X                &kp C                   &kp V   &kp B      &thumb LAYER_Function ESC                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER   &kp N                    &kp M             &kp COMMA              &kp DOT                &kp FSLH             &kp RSHFT
+- &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji ESC          &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number DELETE    &lower    &lower  &thumb LAYER_Mouse TAB  &space LAYER_Symbol SPACE          &thumb LAYER_World ENTER              &kp LBKT             &kp RBKT                &kp END  &magic LAYER_Magic 0
++            &kp LSHFT               &kp Z               &kp X                &kp C                   &kp V   &kp B      &thumb LAYER_Function DELETE                      &kp UP  &kp DOWN  &kp LEFT               &kp RIGHT  &thumb LAYER_System ENTER   &kp N                    &kp M             &kp COMMA              &kp DOT                &kp FSLH             &kp RSHFT
++ &magic LAYER_Magic 0            &kp HOME           &kp PG_UP            &kp PG_DN  &thumb LAYER_Emoji END          &thumb LAYER_Cursor BACKSPACE  &thumb LAYER_Number ESC    &lower    &lower  &thumb LAYER_Mouse ENTER  &space LAYER_Symbol SPACE          &thumb LAYER_World TAB              &kp LBKT             &kp RBKT                &screen_shot  &magic LAYER_Magic 0
+             >;
+         };
+ 
+@@ -2832,8 +2842,8 @@ macros {
+  &trans  &trans  &trans  &trans  &trans  &trans                                                                               &trans  &trans  &trans  &trans  &trans  &trans
+  &trans  &trans   &kp Y  &trans  &trans  &trans                                                                               &trans  &trans  &trans   &kp W  &trans  &trans
+  &trans   &kp C   &kp I   &kp E   &kp A  &trans                                                                               &trans   &kp H   &kp T   &kp S   &kp N  &trans
+- &trans  &trans  &trans  &trans  &trans  &trans   &kp ESC   &trans             &trans             &trans   &trans    &kp RET  &trans  &trans  &trans  &trans  &trans  &trans
+- &trans  &trans  &trans  &trans  &trans          &kp BSPC  &kp DEL  &tog LAYER_Typing  &tog LAYER_Typing  &kp TAB  &kp SPACE          &trans  &trans  &trans  &trans  &trans
++ &trans  &trans  &trans  &trans  &trans  &trans   &kp DEL   &trans             &trans             &trans   &trans    &kp RET  &trans  &trans  &trans  &trans  &trans  &trans
++ &trans  &trans  &trans  &trans  &trans          &kp BSPC  &kp ESC  &tog LAYER_Typing  &tog LAYER_Typing  &kp RET  &kp SPACE          &trans  &trans  &trans  &trans  &trans
+             >;
+         };
+ 
